src/index.test.ts

```diff
@@ -41,6 +41,34 @@ describe('rateLimit', () => {
     expect(elapsed).toBeLessThan(3000)
     expect(done.length).toEqual(5)
   })
+  // This test is here to ensure that the rejected promise doesn't
+  // throw an UnhandledPromiseRejection exception.
+  test('rejections should bubble up .add', async () => {
+    const limiter = rateLimit(3)
+    // Shorter timeout to make sure that this rejects before the other promises resolve
+    await limiter.add(
+      setTimeout(5).then(() => {
+        throw 'rejectedPromise'
+      })
+    )
+    await limiter.add(setTimeout(10))
```

Comment:
You can use [`not.toThrow` and `toThrow`](https://jestjs.io/docs/expect#tothrowerror) for more clarity on these tests.

Reply:
I'm not sure that will work here since what you're testing is that you don't get an UnhandledPromiseRejection exception which by nature exits the Node process.

Reply:
Jest catches the error

```javascript
// From their docs
 expect(() => {
    drinkFlavor('octopus');
  }).toThrow();
```

Unless you're saying even with a try/catch the error crashes node?

Reply:
Yes, that's the issue I'm addressing. The child promise did not have an onRejected handler and the parent promise is not always awaited. At least one of those scenarios would need to be true, otherwise an UnhandledPromiseRejection exception will be thrown and Node will exit the process. See this Runkit for more info: https://runkit.com/dubiousdavid/handling-promise-rejections